taskfiles/lint-cpp.yaml

```
@@ -55,38 +53,37 @@ tasks:
     aliases:
       - "cpp-static-fix"
     desc: "Runs the C++ static analyzers. Only checks for warnings and violations."
-    sources:
+    sources: &cpp_static_src_files
       - "{{.G_CMAKE_CACHE}}"
       - "{{.G_COMPILE_COMMANDS_DB}}"
       - "{{.G_LINT_VENV_CHECKSUM_FILE}}"
-      - "{{.G_CPP_SRC_DIR}}/**/*.cpp"
-      - "{{.G_CPP_SRC_DIR}}/**/*.h"
-      - "{{.G_CPP_SRC_DIR}}/**/*.hpp"
+      - "{{.G_CPP_SRC_DIR}}"
       - "{{.ROOT_DIR}}/.clang-tidy"
       - "{{.TASKFILE}}"
     deps:
       - ":build:init"
       - "cpp-configs"
       - "venv"
     cmds:
-      - task: ":utils:clang-tidy"
+      - task: ":utils:cpp-lint:clang-tidy-find"
         vars:
-          FLAGS: >-
-            --config-file "{{.ROOT_DIR}}/.clang-tidy"
-            -p "{{.G_COMPILE_COMMANDS_DB}}"
-          SRC_PATHS:
-            ref: ".G_LINT_CPP_DIRS"
+          FLAGS:
+            - "--config-file '{{.ROOT_DIR}}/.clang-tidy'"
+            - "-p '{{.G_COMPILE_COMMANDS_DB}}'"
+          INCLUDE_PATTERNS: ["*.cpp", "*.h", "*.hpp", "*.inc"]
```

```suggestion
          INCLUDE_PATTERNS: ["{{.G_CPP_SRC_DIR}}/*"]
```

`INCLUDE_PATTERNS` is different than `INCLUDE_FILENAME_PATTERNS`.

---

Sure, but that's already the default in `clang-tidy-find`, so do we need to specify it?